### PR TITLE
Add GitHub Actions CI/CD Run for liblog

### PR DESCRIPTION
**Pull request by folkengine**
Integrates a basic CI/CD GitHub Action that includes building, lint checks and validating unit tests. While this PR does add a tests folder, it does not include any functional tests.

### CODE REVIEW

**Comment by folkengine** [comment]
@shankari please review.

File: .github/workflows/build_and_test.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Please reference work here https://github.com/EVerest/everest-core/tree/main/.github/workflows
+# TODO: modify to reuse the above workflow to DRY up CI.
+
+name: Build and test liblog
+on: 
+  push:
+    branch: "**"
+  pull_request: {}
```
Comment (shankari) [inline .github/workflows/build_and_test.yaml:8]: you are running this on all pushes but no pull requests?
I believe that `pull_request: {}` means no branches are specified for the pull request, which means that it won't ever run on PR. Have you tested this with a pull request?

**Review by shankari — CHANGES_REQUESTED** [review]
Note that this action is not being executed in this repo, but I've verified it at
https://github.com/US-JOET/liblog since it runs on push

But it hasn't yet been verified with a pull request (see below)....

File: .github/workflows/build_and_test.yaml (hunk 1 above)
Comment (shankari) [reply .github/workflows/build_and_test.yaml:8]: Hm. I guess if no branch is specified, then the event matches any branch
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-a-single-event

But I still haven't found documentation indicating that an explicitly specified blank list is the same as "no branch"

Comment (folkengine) [reply .github/workflows/build_and_test.yaml]: Cleaned this up so that it's clearer. I've used this pattern when I want runs on all branches and PRs:
```
on:
  push:
  pull_request:
```
Confirmed that it's run here.

**Comment by andistorm** [comment]
When the pull request is ready for merging, please mark me as a reviewer.

**Review by shankari — APPROVED** [review]

File: .ci/build-kit/install_and_test.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    -B build \
+    -S "$EXT_MOUNT/source" \
+    -G Ninja \
+    -DBUILD_TESTING=ON \
+    -DCMAKE_BUILD_TYPE=Debug \
+    -DCMAKE_INSTALL_PREFIX="$WORKSPACE_PATH/dist"
+
+ninja -j$(nproc) -C build install
+
+ninja -j$(nproc) -C build tests/test
```
Comment (andistorm) [inline .ci/build-kit/install_and_test.sh:15]:
```suggestion
ninja -j$(nproc) -C build test
```

Unit tests should be "registered" by CTest and ran by the target `test`

File: .github/workflows/build_and_test.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  workflow_dispatch:
+    inputs:
+      runner:
+        description: Which runner to use
+        type: choice
+        default: 'ubuntu-22.04'
+        required: true
+        options:
+          - 'ubuntu-22.04'
+          - 'large-ubuntu-22.04-xxl'
```
Comment (andistorm) [inline .github/workflows/build_and_test.yaml:17]: Please have a look at https://github.com/EVerest/everest-core/blob/main/.github/workflows/build_and_test.yaml#L20 and use the defined input

File: CMakeLists.txt [hunk 4]
```diff
@@ -69,6 +69,8 @@ endif()
 # testing
 if(BUILD_TESTING)
```
Comment (andistorm) [inline CMakeLists.txt:70]: Please have a look at [How do I name test targets in CMake?](https://everest.github.io/nightly/general/faq.html#unit-tests) and adapt the condition in the if statement

File: examples/test_trace.cpp
Comment (andistorm) [inline examples/test_trace.cpp:1]: Does not belong to this PR

File: tests/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,17 @@
+find_package(GTest REQUIRED)
```
Comment (andistorm) [inline tests/CMakeLists.txt:1]: GTest should be included via EDM/CPM. Have a look at https://github.com/EVerest/libevse-security for example

File: tests/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+set(TEST_NAME liblog_test)
```
Comment (andistorm) [inline tests/CMakeLists.txt:3]: test targets names should be prefixed by the project name https://github.com/EVerest/libevse-security

File: tests/liblog_test.cpp [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    void TearDown() override {
+    }
+};
+
+TEST(LibLogUnitTest, test_truth) {
+    ASSERT_TRUE(1 == 1);
+}
+
+} // namespace Logging
+} // namespace Everest
```
Comment (andistorm) [inline tests/liblog_test.cpp:22]:
```suggestion
} // namespace Everest

```

**Review by andistorm — CHANGES_REQUESTED** [review]
Thank you for you contribution and thanks for your patience!

Since this PR was waiting some time a couple of things got best practices. You couldn't know while implementing. This is why I requested a couple of changes.

File: tests/CMakeLists.txt (hunk 5 above)
Comment (folkengine) [reply tests/CMakeLists.txt:1]: @andistorm This library doesn't currently use EDM/CPM. I'm concerned about adding the overhead to this project just to add a test dependency.

Comment (andistorm) [reply tests/CMakeLists.txt:1]: agreed :+1:

File: .github/workflows/build_and_test.yaml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+jobs:
+  lint:
+    name: Lint
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+    runs-on: ${{ matrix.os }}
```
Comment (andistorm) [inline .github/workflows/build_and_test.yaml:24]:
```suggestion
    runs-on: ${{ inputs.runner || 'ubuntu-22.04' }}
```

File: .github/workflows/build_and_test.yaml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+        with:
+          source-dir: source
+          extensions: hpp,cpp
+          exclude: cache
+  install_and_test:
+    name: Install and test
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+    runs-on: ${{ matrix.os }}
```
Comment (andistorm) [inline .github/workflows/build_and_test.yaml:41]:
```suggestion
    runs-on: ${{ inputs.runner || 'ubuntu-22.04' }}
```

**Review by andistorm — CHANGES_REQUESTED** [review]
The suggested chagnes for
> Please have a look at https://github.com/EVerest/everest-core/blob/main/.github/workflows/build_and_test.yaml#L20 and use the defined input

**Review by andistorm — APPROVED** [review]
LGTM, thank you @folkengine :+1:
